src/lib/statistics/chunk_statistics/counting_quotient_filter.cpp

```diff
@@ -0,0 +1,139 @@
+#include "counting_quotient_filter.hpp"
+
+#include <cmath>
+#include <iostream>
+#include <string>
+
+#include "utils/murmur_hash.hpp"
+#include "resolve_type.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "storage/create_iterable_from_segment.hpp"
+#include "types.hpp"
+
+using namespace gqf2; // NOLINT
+using namespace gqf4; // NOLINT
+using namespace gqf8; // NOLINT
+using namespace gqf16; // NOLINT
+using namespace gqf32; // NOLINT
+
+namespace opossum {
+
+template <typename ElementType>
+CountingQuotientFilter<ElementType>::CountingQuotientFilter(uint8_t quotient_size, uint8_t remainder_size) {
+  Assert(quotient_size > 0, "quotient size can not be zero.");
+  Assert(quotient_size + static_cast<uint8_t>(remainder_size) <= 64, "The hash length can not exceed 64 bits.");
+  Assert(remainder_size == 2 || remainder_size == 4 || remainder_size == 8 || remainder_size == 16
+    || remainder_size == 32, "remainder size must be 2, 4, 8, 16, or 32");
+
+  _remainder_size = remainder_size;
+  _number_of_slots = std::pow(2, quotient_size);
+  _hash_bits = quotient_size + remainder_size;
+
+if (remainder_size == 2) {
+    _quotient_filter = gqf2::quotient_filter();

```

Comment:
If you are calling a constructor, I find `{}` clearer

Reply:
Its just initializing the variant with the struct.